### PR TITLE
Split ConnectionPoolValidator for SQLServer and PostgreSQL

### PR DESCRIPTION
**Pull request by jpalac**
Fixes https://github.com/Particular/NServiceBus.SqlServer/issues/1411

### CODE REVIEW

File: src/NServiceBus.Transport.PostgreSql/Configuration/PostgreSqlDbConnectionFactory.cs [hunk 1]
```diff
@@ -7,15 +7,16 @@
     using System.Threading;
     using Npgsql;
     using Sql.Shared.Configuration;
+    using NServiceBus.Transport.Sql.Shared;
 
     class PostgreSqlDbConnectionFactory : DbConnectionFactory
     {
-        public PostgreSqlDbConnectionFactory(Func<CancellationToken, Task<DbConnection>> factory) : base(factory)
+        public PostgreSqlDbConnectionFactory(Func<CancellationToken, Task<DbConnection>> factory, Func<string, ValidationCheckResult> connectionPoolValidator) : base(factory, connectionPoolValidator)
```
Comment (PhilBastian) [inline src/NServiceBus.Transport.PostgreSql/Configuration/PostgreSqlDbConnectionFactory.cs:14]: should this just be named `connectionStringValidator`? The fact that it only checks the connection pool value seems irrelevant here

**Review by PhilBastian — APPROVED** [review]

**Review by andreasohlund — APPROVED** [review]

**Review by SzymonPobiega — APPROVED** [review]
